### PR TITLE
Optimized for perf

### PR DESCRIPTION
**Pull request by SameerK-MSFT**
GetProcess is called only once and formatting is done only once

### CODE REVIEW

File: src/Shared/SharedUtilities.cs [hunk 1]
```diff
@@ -31,6 +32,8 @@ public static class SharedUtilities
 
         private static readonly Lazy<bool> s_isMono = new Lazy<bool>(() => Type.GetType("Mono.Runtime") != null);
 
+        private static readonly Lazy<Process> s_currentProcess = new Lazy<Process>(() => Process.GetCurrentProcess());
```
Comment (pmaytak) [inline src/Shared/SharedUtilities.cs:35]: Hmm, wondering if there are could be issues to staticly hanging onto a process reference? Recommendation [here](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process?view=net-6.0#remarks) is to dispose of Process after being used.

Maybe save a string instead something like this?
```suggestion
        private static readonly Lazy<string> s_currentProcessInfo = new Lazy<string>(() => $"{Process.GetCurrentProcess().Id} {Process.GetCurrentProcess().ProcessName}");
```

File: src/Shared/SharedUtilities.cs [hunk 2]
```diff
@@ -77,6 +80,18 @@ internal static bool IsMonoPlatform()
             return s_isMono.Value;
         }
 
+        /// <summary>
+        /// Instantiates the process if not done already and retrieves the ID the process.
+        /// </summary>
+        /// <returns>process id</returns>
+        public static int GetCurrentProcessId() => s_currentProcess.Value.Id;
```
Comment (pmaytak) [inline src/Shared/SharedUtilities.cs:87]: internal?

**Review by pmaytak — CHANGES_REQUESTED** [review]

File: src/Shared/SharedUtilities.cs (hunk 1 above)
Comment (SameerK-MSFT) [reply src/Shared/SharedUtilities.cs:35]: My thought behind was that this is a static variable. So, the dispose would not be necessary. And keeping it there would help in accessing more properties in the future. I am open either way.

File: src/Shared/SharedUtilities.cs (hunk 2 above)
Comment (SameerK-MSFT) [reply src/Shared/SharedUtilities.cs:87]: This being a utility, public would be good. Anyone can use it.

File: src/Shared/SharedUtilities.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/Shared/SharedUtilities.cs:35]: We want to cache a string, not the Process. Seems less risky with what Peter is proposing.

Comment (SameerK-MSFT) [reply src/Shared/SharedUtilities.cs:35]: I think we can cache id and name separately and create string one time before the loop. So Id and process can be with no context in the shared utility and used later by anyone. Creating formatted string before the loop should not be costly. Also, the format can be changed anytime.

**Review by bgavrilMS — APPROVED** [review]

**Review by gladjohn — APPROVED** [review]
looks good @SameerK-MSFT

**Review by pmaytak — APPROVED** [review]
I'm a bit reserved in this perspective - the only change I would make the Get methods internal, and expose them only if there's a specific need by users.

**Comment by SameerK-MSFT** [comment]
Shows that the pipeline was successful. https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=926323&view=results

**Pull request merged**